### PR TITLE
[METRICS] Remove unnecessary Guava Cache

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

`MetricRegistry` hold counters in `ConcurrentHashMap`, it's unnecessary to use Guava Cache.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsSystem.scala [hunk 1]
```diff
@@ -33,29 +31,17 @@ import org.apache.kyuubi.service.CompositeService
...
-      counter.inc(1L)
-      counter.getCount
-    } finally {
-      countersLock.unlock()
-    }
+
+  def incAndGetCount(key: String): Long = synchronized {
+    val counter = registry.counter(key)
+    counter.inc(1L)
+    counter.getCount
```
Comment (pan3793) [inline kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsSystem.scala:38]: The return value never used, if return `Unit`, we can remove `synchronized`

Comment (yaooqinn) [reply kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsSystem.scala:38]: sgtm

**Review by ulysses-you — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.3.0 and branch 1.2 for v1.2.0
